### PR TITLE
Improve deployment details

### PR DESCRIPTION
**Pull request by MohamedElmdary**
## Issues
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/257
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/258
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/259
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/260
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/261
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/262
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/263
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/264
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/265
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/266
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/267
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/272
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/277

### CODE REVIEW

**Review by zaelgohary — CHANGES_REQUESTED** [review]
Can we remove the public IP switch? It can't be enabled anyway.
https://github.com/threefoldtech/tfgrid-sdk-ts/blob/2853ebd3b37c48dc45a897556d1ed24d969d71ae/packages/playground/src/components/deployment_data_dialog.vue#L25

File: packages/playground/src/components/deployment_data_dialog.vue [hunk 1]
```diff
@@ -120,11 +131,50 @@ function copy() {
...
+function _transform(value: string): any {
+  const v = value.toLowerCase();
+  if (v === "true" || v === "false") {
+    return Boolean(v);
+  }
+  return value;
+}
+
+function getType(key: string): string {
+  if (props.environments === false) {
```
Comment (zaelgohary) [inline packages/playground/src/components/deployment_data_dialog.vue:166]: I don't think this condition is necessary as it returns the same as the default.

**Comment by MohamedElmdary** [comment]
> Can we remove the public IP switch? It can't be enabled anyway.
> 
> https://github.com/threefoldtech/tfgrid-sdk-ts/blob/2853ebd3b37c48dc45a897556d1ed24d969d71ae/packages/playground/src/components/deployment_data_dialog.vue#L25

it indicates that public ip isn't enabled

**Comment by zaelgohary** [comment]
All SMTP inputs will be shown even if no data was provided.

**Review by zaelgohary — APPROVED** [review]

**Pull request merged**